cmd/swarm/swarm-smoke/pss.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/swarm/pss"
+	"github.com/ethereum/go-ethereum/swarm/testutil"
+
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+const (
+	pssModeRaw = iota
+	pssModeAsym
+	pssModeSym
+)
+
+type pssJob struct {
+	msg  []byte
+	mode int
+	prox bool
+}
+
+type pssNode struct {
+	hostIdx        int
+	addr           []byte
+	pubkey         string
+	client         *rpc.Client
+	deregisterFunc func()
+	msgC           chan pss.APIMsg
+}
+
+// keyed by hex representation of overlay address
+type pssSession struct {
+	topic        pss.Topic
+	msgC         chan pss.APIMsg
+	nodes        []*pssNode
+	requiredJobs map[string]*pssJob
+	allowedJobs  map[string]*pssJob
+}
+
+func pssChecks(ctx *cli.Context, tuid string) error {
+	return pssAsym(ctx, tuid)
+}
+
+func pssAsym(ctx *cli.Context, tuid string) error {
```

Comment:
can we please have a brief text describing what we are testing, what are the test's erroneous stop conditions (if they at all exist), does the test have a green exit condition? or does it just go on running till it fails?